ProcessLib/SmallDeformation/SmallDeformationProcess.h

```diff
@@ -171,6 +196,48 @@ class SmallDeformationProcess final : public Process
             dxdot_dx, dx_dx, M, K, b, Jac, coupling_term);
     }
 
+    void setInitialConditionsConcreteProcess(double const t,
+                                             GlobalVector const& x) override
+    {
+        DBUG("SetInitialConditions SmallDeformationProcess.");
+
+        if (!_mesh.getProperties().hasPropertyVector("integration_point_data"))
+            return;
+        if (!_mesh.getProperties().hasPropertyVector("integration_point_offsets"))
+            OGS_FATAL(
+                "integration_point_data field exists in the input but there is "
+                "no integration_point_offsets cell data.");
+
+        auto const& data =
+            *_mesh.getProperties().template getPropertyVector<char>(
+                "integration_point_data");
+        assert(data.getMeshItemType() ==
+               MeshLib::MeshItemType::IntegrationPoint);
+
+        auto const& offsets =
+            *_mesh.getProperties().template getPropertyVector<std::size_t>(
+                "integration_point_offsets");
+        assert(offsets.getMeshItemType() == MeshLib::MeshItemType::Cell);
+
+        std::vector<char> local_data;
+        assert(_local_assemblers.size() == offsets.size());
+        // Starting counting from one; the last cell is handled after the loop.
+        std::size_t i = 0;
+        for (; i < _local_assemblers.size() - 1; ++i)
+        {
+            std::size_t const size = offsets[i + 1] - offsets[i];
+            local_data.resize(size);
+            std::memcpy(local_data.data(), &data[offsets[i]], size);
```

Comment:
Minor: Copy could be avoided, but then you cannot pass a std::vector to `readIntegrationPointData()`.